File: backend/pigeon_app/models/adventure.py
from django.db import models
from .player import Player
from rest_framework import serializers


class Adventure(models.Model):
    player = models.ForeignKey(Player, on_delete=models.CASCADE)
    lvl = models.IntegerField(default=0)
    encounter = models.IntegerField(default=0)
    nb_tries = models.IntegerField(default=0)
    is_success = models.BooleanField(default=False)
    reward_droppings = models.IntegerField(default=0)
    created_at = models.DateTimeField(auto_now_add=True, null=True)
    completed_at =  models.DateTimeField(null=True)

class AdventureSerializer(serializers.ModelSerializer):
    class Meta:
        model = Adventure
        fields = '__all__'

File: frontend/src/app/core/models/adventure.ts
import * as internal from "events"
import { Player } from "./player"

export interface Adventure {
    player : number
    lvl : number
    encounter : number
    nb_tries : number
    is_success : false
    reward_droppings : 0
    created_at : Date
    completed_at : Date
}
